### ISSUE TITLE
[Campaign] [Units] [The South Guard] Update Deoran's custom unit-line.

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#6815]
### Describe the desired feature

The current graphics for Deoran's unit-line (Junior Commander -> Knight Commander -> Mounted General) are very outdated. Should be updated in-line to match the current Loyalist mounted units.

These have been offered by the forum user: [Angelonius](https://forums.wesnoth.org/memberlist.php?mode=viewprofile&u=142256)
(They uploaded these in the #modding channel of the official discord months ago)

![junior_commander](https://user-images.githubusercontent.com/5283677/176689066-db79c9eb-b70b-49e6-b97d-4424fd9d220f.png)  
Level 1 - Junior Commander

![knight_commander](https://user-images.githubusercontent.com/5283677/176689088-59dbccb2-8f70-47a6-8743-85e9c8a774e7.png) 
Level 2 - Knight Commander

![mounted_general](https://user-images.githubusercontent.com/5283677/176689105-b9d044e7-eea3-4c50-bc19-3adcad3008a7.png) 
Level 3 - Mounted General

Maybe these can be used? Or, adapted as a base?
Pinging @nemaara and @doofus-01 here for remarks.

**Comment by doofus-01** [comment]
Ideally, there would be something a little more distinctive, but I agree the current sprites are outdated.  If those have basic animations (defense, at least), I've got no objections to replacement.

**Comment by cooljeanius** [comment]
I don't think that being outdated is a good enough reason to call for a replacement. The existing sprites are fine.

**Comment by knyghtmare** [comment]
I would use "not homogenous anymore to the other Loyalists mounted units" as a second reason. Compare with the current Cavalryman and Horseman unit trees. Deoran looks out of place now. If this was still 1.6, then the current ones could have stayed, but it's the eve of 1.18 and the campaign only units could use some attention for once at least.

**Comment by knyghtmare** [comment]
Closed by #9682 
Hats off to @Dalas121 for actually making the Deoran have custom units in both TSG and TDG.
